### ISSUE TITLE
Spel failed with environment.getProperty(java.lang.String,java.lang.String)

### ISSUE THREAD

**Issue by linux-china** [spring-projects-experimental/spring-native#768]
I did a demo with Spring Native with gRPC, and I use popular https://github.com/LogNet/grpc-spring-boot-starter.  In GRpcAutoConfiguration.java and there is a Spel expression as following:    

```java
     @Bean
    @ConditionalOnExpression("#{environment.getProperty(\'grpc.inProcessServerName\',\'\')!=\'\'}")
    public GRpcServerRunner grpcInprocessServerRunner(@Qualifier("grpcInternalConfigurator") Consumer<ServerBuilder<?>> configurator) {
        return new GRpcServerRunner(configurator, InProcessServerBuilder.forName(grpcServerProperties.getInProcessServerName()));
    }
```

And I added  `<removeSpelSupport>false</removeSpelSupport>` for the spring-aot-maven-plugin.  

Native image compile is all good, and after start of native image app, and I get the following error.  

```
2021-05-04 11:38:46.936  INFO 50889 --- [           main] com.example.SpringBootApp                : Starting SpringBootApp using Java 11.0.11 on linux-chinadeMacBook-Pro.local with PID 50889 (/Users/linux_china/IdeaProjects/grpc-native-demo/target/grpc-native-demo started by linux_china in /Users/linux_china/IdeaProjects/grpc-native-demo)
2021-05-04 11:38:46.937  INFO 50889 --- [           main] com.example.SpringBootApp                : No active profile set, falling back to default profiles: default
2021-05-04 11:38:46.961 ERROR 50889 --- [           main] o.s.boot.SpringApplication               : Application run failed

java.lang.IllegalStateException: Error processing condition on org.lognet.springboot.grpc.autoconfigure.GRpcAutoConfiguration.grpcInprocessServerRunner
	at org.springframework.boot.autoconfigure.condition.SpringBootCondition.matches(SpringBootCondition.java:60) ~[grpc-native-demo:2.4.5]
	at org.springframework.context.annotation.ConditionEvaluator.shouldSkip(ConditionEvaluator.java:108) ~[na:na]
	at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.loadBeanDefinitionsForBeanMethod(ConfigurationClassBeanDefinitionReader.java:193) ~[na:na]
	at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.loadBeanDefinitionsForConfigurationClass(ConfigurationClassBeanDefinitionReader.java:153) ~[na:na]
	at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.loadBeanDefinitions(ConfigurationClassBeanDefinitionReader.java:129) ~[na:na]
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.processConfigBeanDefinitions(ConfigurationClassPostProcessor.java:343) ~[grpc-native-demo:5.3.6]
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.postProcessBeanDefinitionRegistry(ConfigurationClassPostProcessor.java:247) ~[grpc-native-demo:5.3.6]
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanDefinitionRegistryPostProcessors(PostProcessorRegistrationDelegate.java:311) ~[na:na]
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:112) ~[na:na]
	at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:746) ~[na:na]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:564) ~[na:na]
	at org.springframework.boot.web.reactive.context.ReactiveWebServerApplicationContext.refresh(ReactiveWebServerApplicationContext.java:63) ~[na:na]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:782) ~[grpc-native-demo:2.4.5]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:774) ~[grpc-native-demo:2.4.5]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:439) ~[grpc-native-demo:2.4.5]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:339) ~[grpc-native-demo:2.4.5]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1340) ~[grpc-native-demo:2.4.5]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1329) ~[grpc-native-demo:2.4.5]
	at com.example.SpringBootApp.main(SpringBootApp.java:9) ~[grpc-native-demo:na]
Caused by: org.springframework.beans.factory.BeanExpressionException: Expression parsing failed; nested exception is org.springframework.expression.spel.SpelEvaluationException: EL1004E: Method call: Method getProperty(java.lang.String,java.lang.String) cannot be found on type org.springframework.boot.web.reactive.context.StandardReactiveWebEnvironment
	at org.springframework.context.expression.StandardBeanExpressionResolver.evaluate(StandardBeanExpressionResolver.java:169) ~[na:na]
	at org.springframework.boot.autoconfigure.condition.OnExpressionCondition.evaluateExpression(OnExpressionCondition.java:60) ~[grpc-native-demo:2.4.5]
	at org.springframework.boot.autoconfigure.condition.OnExpressionCondition.getMatchOutcome(OnExpressionCondition.java:48) ~[grpc-native-demo:2.4.5]
	at org.springframework.boot.autoconfigure.condition.SpringBootCondition.matches(SpringBootCondition.java:47) ~[grpc-native-demo:2.4.5]
	... 18 common frames omitted
Caused by: org.springframework.expression.spel.SpelEvaluationException: EL1004E: Method call: Method getProperty(java.lang.String,java.lang.String) cannot be found on type org.springframework.boot.web.reactive.context.StandardReactiveWebEnvironment
	at org.springframework.expression.spel.ast.MethodReference.findAccessorForMethod(MethodReference.java:226) ~[na:na]
	at org.springframework.expression.spel.ast.MethodReference.getValueInternal(MethodReference.java:135) ~[na:na]
	at org.springframework.expression.spel.ast.MethodReference.access$000(MethodReference.java:55) ~[na:na]
	at org.springframework.expression.spel.ast.MethodReference$MethodValueRef.getValue(MethodReference.java:387) ~[na:na]
	at org.springframework.expression.spel.ast.CompoundExpression.getValueInternal(CompoundExpression.java:92) ~[na:na]
	at org.springframework.expression.spel.ast.OpNE.getValueInternal(OpNE.java:43) ~[na:na]
	at org.springframework.expression.spel.ast.OpNE.getValueInternal(OpNE.java:33) ~[na:na]
	at org.springframework.expression.spel.ast.SpelNodeImpl.getValue(SpelNodeImpl.java:112) ~[na:na]
	at org.springframework.expression.spel.standard.SpelExpression.getValue(SpelExpression.java:272) ~[na:na]
	at org.springframework.context.expression.StandardBeanExpressionResolver.evaluate(StandardBeanExpressionResolver.java:166) ~[na:na]
	... 21 common frames omitted
```

The demo is  on https://github.com/linux-china/grpc-native-demo and native build command is `https://github.com/linux-china/grpc-native-demo`

**Comment by sdeleuze** [comment]
Maybe the method `StandardReactiveWebEnvironment#getProperty(String,String)` invoked via Spel is removed since not visible by the static analysis. Could you try to add related hint `@TypeHint(types = StandardReactiveWebEnvironment.class, access = AccessBits.DECLARED_METHODS)` and see how it goes?

**Comment by spring-projects-issues** [comment]
If you would like us to look at this issue, please provide the requested information. If the information is not provided within the next 7 days this issue will be closed.

**Comment by linux-china** [comment]
@sdeleuze  it does not work with `@TypeHint(types = StandardReactiveWebEnvironment.class, access = AccessBits.DECLARED_METHODS)` .  stacktrace as following: 


```
Caused by: org.springframework.beans.factory.BeanExpressionException: Expression parsing failed; nested exception is org.springframework.expression.spel.SpelEvaluationException: EL1004E: Method call: Method getProperty(java.lang.String,java.lang.String) cannot be found on type org.springframework.boot.web.reactive.context.StandardReactiveWebEnvironment
	at org.springframework.context.expression.StandardBeanExpressionResolver.evaluate(StandardBeanExpressionResolver.java:169) ~[na:na]
	at org.springframework.boot.autoconfigure.condition.OnExpressionCondition.evaluateExpression(OnExpressionCondition.java:60) ~[grpc-native-demo:2.4.5]
	at org.springframework.boot.autoconfigure.condition.OnExpressionCondition.getMatchOutcome(OnExpressionCondition.java:48) ~[grpc-native-demo:2.4.5]
	at org.springframework.boot.autoconfigure.condition.SpringBootCondition.matches(SpringBootCondition.java:47) ~[grpc-native-demo:2.4.5]
	... 18 common frames omitted
Caused by: org.springframework.expression.spel.SpelEvaluationException: EL1004E: Method call: Method getProperty(java.lang.String,java.lang.String) cannot be found on type org.springframework.boot.web.reactive.context.StandardReactiveWebEnvironment
	at org.springframework.expression.spel.ast.MethodReference.findAccessorForMethod(MethodReference.java:226) ~[na:na]
	at org.springframework.expression.spel.ast.MethodReference.getValueInternal(MethodReference.java:135) ~[na:na]
	at org.springframework.expression.spel.ast.MethodReference.access$000(MethodReference.java:55) ~[na:na]
	at org.springframework.expression.spel.ast.MethodReference$MethodValueRef.getValue(MethodReference.java:387) ~[na:na]
	at org.springframework.expression.spel.ast.CompoundExpression.getValueInternal(CompoundExpression.java:92) ~[na:na]
	at org.springframework.expression.spel.ast.OpNE.getValueInternal(OpNE.java:43) ~[na:na]
	at org.springframework.expression.spel.ast.OpNE.getValueInternal(OpNE.java:33) ~[na:na]
	at org.springframework.expression.spel.ast.SpelNodeImpl.getValue(SpelNodeImpl.java:112) ~[na:na]
	at org.springframework.expression.spel.standard.SpelExpression.getValue(SpelExpression.java:272) ~[na:na]
	at org.springframework.context.expression.StandardBeanExpressionResolver.evaluate(StandardBeanExpressionResolver.java:166) ~[na:na]
	... 21 common frames omitted
```

**Comment by sdeleuze** [comment]
Ok, so indeed the right hints should be that because `getProperty(String, String)` is in `AbstractEnvironment` not `StandardReactiveWebEnvironment`:
```
@TypeHint(types = StandardReactiveWebEnvironment.class)
@TypeHint(types = AbstractEnvironment.class, methods = @MethodHint(name = "getProperty", parameterTypes = {String.class, String.class}))
```

This issue will be solved without additional configuration by #167.
